server/sonar-server/src/main/java/org/sonar/server/computation/issue/IssueAssigner.java

```diff
@@ -105,64 +97,16 @@ public void afterComponent(Component component) {
-      BatchReport.Changesets.Changeset changeset = scmChangesets.getChangeset(scmChangesets.getChangesetIndexByLine(line - 1));
-      author = changeset.hasAuthor() ? changeset.getAuthor() : null;
-    }
-    return StringUtils.defaultIfEmpty(author, lastCommitAuthor);
-  }
-
-  private BatchReport.Changesets loadScmChangesetsFromReport(Component component) {
-    return reportReader.readChangesets(component.getReportAttributes().getRef());
-  }
-
-  private BatchReport.Changesets loadScmChangesetsFromDb(Component component) {
-    DbSession dbSession = dbClient.openSession(false);
-    try {
-      Optional<Iterable<DbFileSources.Line>> lines = sourceService.getLines(dbSession, component.getUuid(), 1, Integer.MAX_VALUE);
-      Map<String, BatchReport.Changesets.Changeset> changesetByRevision = new HashMap<>();
-      BatchReport.Changesets.Builder scmBuilder = BatchReport.Changesets.newBuilder()
-        .setComponentRef(component.getReportAttributes().getRef());
-      if (lines.isPresent()) {
-        for (DbFileSources.Line sourceLine : lines.get()) {
-          String scmRevision = sourceLine.getScmRevision();
-          if (scmRevision == null || changesetByRevision.get(scmRevision) == null) {
-            BatchReport.Changesets.Changeset.Builder changeSetBuilder = BatchReport.Changesets.Changeset.newBuilder();
-            if (sourceLine.hasScmAuthor()) {
-              changeSetBuilder.setAuthor(sourceLine.getScmAuthor());
-            }
-            if (sourceLine.hasScmDate()) {
-              changeSetBuilder.setDate(sourceLine.getScmDate());
-            }
-            if (scmRevision != null) {
-              changeSetBuilder.setRevision(scmRevision);
-            }
-
-            BatchReport.Changesets.Changeset changeset = changeSetBuilder.build();
-            scmBuilder.addChangeset(changeset);
-            scmBuilder.addChangesetIndexByLine(scmBuilder.getChangesetCount() - 1);
-            if (scmRevision != null) {
-              changesetByRevision.put(scmRevision, changeset);
-            }
-          } else {
-            scmBuilder.addChangesetIndexByLine(scmBuilder.getChangesetList().indexOf(changesetByRevision.get(scmRevision)));
-          }
-        }
-      }
-      return scmBuilder.build();
-    } finally {
-      dbClient.closeSession(dbSession);
-    }
-  }
-
-  private void computeLastCommitDateAndAuthor() {
-    for (BatchReport.Changesets.Changeset changeset : scmChangesets.getChangesetList()) {
-      if (changeset.hasAuthor() && changeset.hasDate() && changeset.getDate() > lastCommitDate) {
-        lastCommitDate = changeset.getDate();
-        lastCommitAuthor = changeset.getAuthor();
+    if (line != null && scmChangesets != null) {
+      if (scmChangesets.existsForLine(line)) {
+        author = scmChangesets.getForLine(line).getAuthor();
+      } else {
+        LOGGER.warn("Issue {} is set on a none existing line", issue);
```

Comment:
the fact there is no changeset does not mean the line does not exist. I think the log is incorrect